nft/ethereum_payment_obligation.go

```diff
@@ -287,3 +293,26 @@ func getDocumentService(documentType string) (documents.Service, error) {
 	}
 	return docService, nil
 }
+
+// getCollaborator returns the needed collaboratorField for a PaymentObligation NFT
+// In the current contract implementation the proofField for collaborator is a separated parameter
+// pattern: 'collaborators' + '[i]'
+// examples: 'collaborators[0]','collaborators[1]', etc
+func getCollaboratorProofField(proofFields []string) (string, error) {
+
+	for _, proofField := range proofFields {
+
+		match, err := regexp.MatchString("collaborators\\[[0-9]*\\]", proofField)
```

Comment:
Can we use this instead `collaborators\[[0-9]+\]` ?

This will avoid the edge case `collaborators[]`

Reply:
good catch. I thought about it yesterday to change it.